feat: custom rejection message documentation. (#625)
kubewarden/docs: docs/howtos/policies.md

    modes:
      enforce: "restricted"
```

## Custom rejection message

> When a policy rejects a resource, the error message shown to the user is, by
> default, the error message returned by the policy. However, sometimes cluster
> operators may want to set their own custom reject messages, providing
> instructions to users. This can be done using the `message` field in the
> `ClusterAdmissionPolicy` and `AdmissionPolicy` types.

Reviewer: ```suggestion
When a policy rejects a resource, the error message shown to the user is
the error message returned by the policy. Sometimes cluster
operators may want to set custom reject messages, providing
instructions to users. You can do this using the `message` field in the
`ClusterAdmissionPolicy` and `AdmissionPolicy` types.
```

> :::note
> `ClusterAdmissionPolicyGroup` and `AdmissionPolicyGroup` also have the
> `message` field. But these policies types already uses it as rejection message
> by design.
> :::

Reviewer: ```suggestion
:::note
`ClusterAdmissionPolicyGroup` and `AdmissionPolicyGroup` also have the
`message` field. These policy types already use the `message` field as a rejection message.
:::
```

> The `message` field allows cluster operators to define a custom reject message
> that will override the message returned by the policy. When this configuration
> is set, the original reject message returned by the policy will be stored in the
> `causes` field of the response.

Reviewer: ```suggestion
The `message` field lets cluster operators define a custom reject message
overriding the message returned by the policy. When using this configuration,
the original reject message returned by the policy is stored in the
`causes` field of the response.
```

```yaml
apiVersion: policies.kubewarden.io/v1
kind: ClusterAdmissionPolicy
metadata:
  name: pod-privileged-with-message
spec:
  module: registry://ghcr.io/kubewarden/policies/pod-privileged:latest
  policyServer: default
  mode: protect
  message: "Nops! You cannot do that"
  settings: {}
  rules:
    - apiGroups:
        - ""
      apiVersions:
        - v1
      resources:
        - "*"
      operations:
        - CREATE
  mutating: false
```

> When a privileged pod is deployed in the cluster, the user will get
> the following feedback:

Reviewer: ```suggestion
When you deploy a privileged pod in the cluster, the user gets
tthis feedback:
```
